runtime/lua/vim/lsp/protocol.lua

```diff
@@ -15,7 +15,6 @@ local sysname = vim.uv.os_uname().sysname
 --- @class vim.lsp.protocol.constants
 --- @nodoc
 local constants = {
-  --- @enum lsp.DiagnosticSeverity
```

Comment:
Why are we removing these annotations? Won't this make the enum types no longer visible outside the repo?

Reply:
These are already defined in the generated meta file. These were causing duplicated definition warnings.

Reply:
I think we should then remove the definitions using `@alias` in the meta file and keep these as they are. The enum would provide better completion support.

Reply:
Or actually I'm fine with this for now. I'll refactor it later.

Reply:
Opened https://github.com/neovim/neovim/pull/32215 to address this. It will conflict with this PR, but I think this will get merged first so I can rebase later.

Reply:
I don't mind either way. I'll rebase this before I merge anyway.